tests/core/limits_test.py

```diff
@@ -59,6 +59,19 @@ def test_find_limits_no_finite_values_raises():
         _ = find_limits(x)
 
 
+def test_find_limits_all_zeros():
+    x = sc.zeros(sizes={'x': 5}, unit='s')
+    lims = find_limits(x)
+    assert sc.identical(lims[0], sc.scalar(0.0, unit='s'))
+    assert sc.identical(lims[1], sc.scalar(0.0, unit='s'))
+
+
+def test_find_limits_all_zeros_log_raises():
+    x = sc.zeros(sizes={'x': 5}, unit='s')
+    with pytest.raises(ValueError, match="No positive values were found in array"):
+        _ = find_limits(x, scale='log')
```

Comment:
What is the effect of this change on an actual plot? What happens when toggling the 'log' button on such a plot?

Reply:
Yeah it's a good point. The first version of the fix I wrote just set the limits to `[0.1, 1]` instead of raising an error.
Then, when writing the docstring, I thought it should be consistent with the other case where there are no finite values in the input, so I changed to raise an error.

But you are right that if you have a plot with just zeros, and then you click 'log' then it doesn't update the plot, but just raises an error instead. So I've gone back to `[0.1, 1]`

Reply:
Would it be an option to disable the button?

Reply:
That would connect too many things together I feel.

What do you do if you have an interactive plot with a slider, and on the first slice, you can see data, and you have set `norm='log'`.
Then you move to the second slice where all values are zero. You can't disable the button because you are already in log, and the plot will just raise an error and the position of the slider will be out if sync with the curve displayed on the canvas becase it won't have updated and will still show the first slice.

I think the default limits is not the cleanest but it's what will give the less headaches in practise. It is also simple, and does not require communication between the canvas and the toolbar, when new values are sent to update the curve.